lib/ain-evm/src/core.rs

```diff
@@ -816,6 +816,91 @@ impl EVMCoreService {
 
         Ok((listener.trace, execution_success, return_value, used_gas))
     }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn create_access_list(&self, arguments: EthCallArgs) -> Result<(AccessList, u64)> {
+        let EthCallArgs {
+            caller,
+            to,
+            value,
+            data,
+            gas_limit,
+            access_list,
+            block_number,
+            ..
+        } = arguments;
+
+        let block_header = self
+            .storage
+            .get_block_by_number(&block_number)?
+            .ok_or_else(|| format_err!("Block not found"))
+            .map(|block| block.header)?;
+        let state_root = block_header.state_root;
+        debug!(
+            "Calling EVM at block number : {:#x}, state_root : {:#x}",
+            block_number, state_root
+        );
+
+        let vicinity = Vicinity::from(block_header);
+        let mut backend = EVMBackend::from_root(
+            state_root,
+            Arc::clone(&self.trie_store),
+            Arc::clone(&self.storage),
+            vicinity.clone(),
+            None,
+        )
+        .map_err(|e| format_err!("Could not restore backend {}", e))?;
+        backend.vicinity.origin = caller;
+        backend.vicinity.gas_price = arguments.gas_price;
+
+        static CONFIG: Config = Config::shanghai();
```

Comment:
We should look into sharing this.
We could make AinExecutor CONFIG pub and  use `static CONFIG: Config = AinExecutor::CONFIG;` or use a single global const

Reply:
Addressed in [117b9cd](https://github.com/DeFiCh/ain/pull/2770/commits/117b9cd26aa25a4b8813f4b967670eb759cb2b28)